R/zchunk_LA124.heat.R

```diff
@@ -36,74 +37,186 @@ module_energy_LA124.heat_DISABLED <- function(command, ...) {
     A24.globaltech_coef <- get_data(all_data, "energy/A24.globaltech_coef")
     calibrated_techs <- get_data(all_data, "energy/calibrated_techs")
     enduse_fuel_aggregation <- get_data(all_data, "energy/enduse_fuel_aggregation")
-    L1011.en_bal_EJ_R_Si_Fi_Yh <- get_data(all_data, "temp-data-inject/L1011.en_bal_EJ_R_Si_Fi_Yh")
-    L1231.out_EJ_R_elec_F_tech_Yh <- get_data(all_data, "temp-data-inject/L1231.out_EJ_R_elec_F_tech_Yh")
+
+    get_data(all_data, "temp-data-inject/L1011.en_bal_EJ_R_Si_Fi_Yh") %>%
+      gather(year, value, -fuel, -sector, -GCAM_region_ID) %>%
+      mutate(year = sub("X", "", year)) %>%
+      mutate(year = as.integer(year)) -> L1011.en_bal_EJ_R_Si_Fi_Yh
+
+    L1231.out_EJ_R_elec_F_tech_Yh <- get_data(all_data, "L1231.out_EJ_R_elec_F_tech_Yh")
 
     # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses vecpaste
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # ===================================================
+    # Create list of heat regions
+    A_regions %>%
+      filter(heat == 1) %>%
```

Comment:
It looks like heat is a logical variable to determine which regions have district heat. Not sure how A_regions.csv, which contains several other information, is created in regards to identifying regions with district heat. I think it's done manually, but I wonder if it can be done using the IEA energy balance data which the determination must be based on. I would recommend, replacing "heat" in A_regions.csv to "has_district_heat", and then use filter( has_district_heat == 1 ) above.